src/clickonce/launcher/HostFinder.cs

```diff
@@ -28,10 +31,48 @@ internal static class HostFinder
         /// 3) Global location
         /// </summary>
         /// <param name="applicationFilePath">Full path to application</param>
-        /// <returns></returns>
+        /// <returns>Path to host</returns>
         internal static string GetHost(string applicationFilePath)
         {
-            string arch = GetProcessorArchitectureFromAssembly(applicationFilePath);
+            HostFinder hf = new HostFinder(applicationFilePath);
+            return hf.GetHost();
```

Comment:
> There doesn't seem to be much value added by making this into an instance type. You only ever use it from this static method. If I look through the entire call graph the only state used after construction is `arch`, which seems to mean `applicationTargetArchitecture`.

Correct. I tried to avoid adding unnecessary parameters to a handful of methods. I'll see if I can make this cleaner and get rid of GetX86Host/Get64bitHost.

Reply:
Fixed this in new commit.